packages/@mantine/dates/src/components/DateInput/DateInput.tsx

```diff
@@ -167,7 +167,7 @@ export const DateInput = factory<DateInputFactory>((_props, ref) => {
 
   useEffect(() => {
     if (controlled) {
-      setDate(value!);
+      setDate(value ? value! : minDate!);
```

Comment:
When `clearable: true` and you remove a selected value, visually it's removed from the UI however `_date` still gets set to the `minDate` instead of `null`.